### DISCUSSION TITLE
Unmounting does not clear errors and is still included in isValid check (after `@6.10.1`)

### DISCUSSION THREAD

**Discussion by changyeamoon** [react-hook-form/react-hook-form#3523] [Q&A]
**I'm just looking for general guidance**

I'm only looking for general guidance because I can't reproduce it in a sandbox. But I've spent hours trying to figure out why its happening in my local environment. I've mimicked all my dependency versions, mimicked the exact way i'm using `react-hook-form`. It seems to work fine with version `@6.10.1` but any version after that does not work

https://codesandbox.io/s/react-hook-form-isvalid-bug-8yszv?file=/src/index.js

The sandbox above is the basic scenario, I have a radio button that conditionally renders another form input. **It works as expected in the sandbox**, but locally it seems to still think that the extra fields still exist when you remove them. You can see it more clearly when you cause an error in the extra fields, and then switch the radio button to 'male', it will remove the errors **as expected**. But in my codebase this doesn't happen (after `@6.10.1`). I'm stumped and sound crazy cause it seems to work fine in any version in the code sandbox.

Any help would be appreciated. I'll keep trying to reproduce for the time being

**Comment by changyeamoon** [comment]
@bluebill1049 I did not post under `issues` as I can't reproduce it in the sandbox. but im hoping you might have insight into why this might be happening or where I should more closely look.

It's really strange to me that it only happens to me when i upgrade the package from `@6.10.01`

I also need version `@6.11.0` for a bug when removing field array fields. so staying at `@6.10.01` is a no go for me

**Reply by bluebill1049** [reply]
hmmm, that's strange @changyeamoon, everything looks fine to me at your CSB, but i know there is something changed to cause this. so you can confirm 6.10 works and breaks at 6.10.1? maybe i can focus on some of the commits which may lead to this.

**Reply by changyeamoon** [reply]
It works at `6.10.1` and breaks at `6.11.0` + I didnt check the beta versions

i'll keep cracking at reproducing

**Reply by bluebill1049** [reply]
so after 6.10.1, all broken? maybe it's 6.10.2 introduce it then?

**Reply by bluebill1049** [reply]
only single patch tho: https://github.com/react-hook-form/react-hook-form/releases/tag/v6.11.2

**Reply by changyeamoon** [reply]
let me check the beta's real quick

**Reply by bluebill1049** [reply]
Thanks @changyeamoon 🙏

**Reply by changyeamoon** [reply]
@bluebill1049 I just checked ~`6.10.2-beta.1` , `6.10.2-beta.2`~ ,  `6.11.0`, `6.11.1` , `6.11.2` , `6.11.3` , `6.11.4` , `6.11.5` , `6.11.6-beta.1` . all broken

**Reply by changyeamoon** [reply]
@bluebill1049 oops i mean `6.10.2-beta.1` , `6.10.2-beta.2` still works fine

**Reply by bluebill1049** [reply]
oh I see, thanks. It's the minor version.

**Reply by changyeamoon** [reply]
@bluebill1049 how do you mean? Do you know whats wrong o.o

**Reply by bluebill1049** [reply]
I meant it's the release of the minor version (6.1.0) which caused this issue, I haven't been able to reproduce the problem just yet.
